hs628_tools/Visualization.py

```diff
@@ -0,0 +1,294 @@
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+import pandas as pd
+import sys 
+import plotly.plotly as py
+import plotly.graph_objs as go
+#plotly.tools.set_credentials_file(username='hhowell', api_key='BRP1aHhfTHwkqnvalVQL')
+from sklearn import datasets, linear_model
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import mean_squared_error,mean_absolute_error, r2_score
+###This is importing a class from Issue #11 Test module. We used the ratio function to determine the test size ratio for the train_test_split function. 
+#from Test import data_split as ds
+
+
+###Bar Plot-Count
+#This function counts all instances in a variable and creates a barplot to display the results 
+#@param: The variable to be counted, the x_axis label, title of the graph
+#@return: A count bar plot of the variable inputted 
+def count_and_print(variable, x_label, title):
+    def count(variable):
+        new_dic = {}
+        c = (len(variable)-1)
+        for index, value in enumerate(variable):
+            count = 0
+            if((value in new_dic) == True):
+                pass
+            else:
+                for index2, value2 in enumerate(variable):
+                    if value == value2:
+                        count = count + 1
+                    if index2 == (len(variable)-1):
+                        new_dic[value] =  count
+        return new_dic
+
+    variable = variable
+
+    count_dictionary = count(variable)
+    from collections import OrderedDict
+    from operator import itemgetter
+
+    count_ordered = (OrderedDict(sorted(count_dictionary.items(), key = itemgetter(0), reverse = False)))
+
+    def key_list(list):
+        keys_list= []
+        for key in list.keys():
+            keys_list.append(key)
+        return keys_list
+
+    def value_list(list):
+        values_list= []
+        for value in list.values():
+            values_list.append(value)
+        return values_list
+    keys = key_list(count_ordered)
+    values = value_list(count_ordered)
+#Plot the variable
+    f1 = (go.Bar(
+                x=keys,
+                y=values
+        )
+           )
+    layout = go.Layout(
```

Comment:
this ```layout =``` line may be too long for a single line, but the current formatting seems too disperse.